src/SubPool.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.26;
+
+import {Auth} from "./Auth.sol";
+import {ISubPoolFactory} from "./interfaces/ISubPoolFactory.sol";
+import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+import {ERC20} from "solady/tokens/ERC20.sol";
+
+using SafeTransferLib for address;
+
+contract SubPool is Auth {
```

Comment:
Missing `receive` function. (This contract is designed to handle ETH but there's no way to deposit ETH into it.)

Reply:
idea was to only deposit more ETH when necessary via `heal` but will add receive too